test_chess.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+from unittest.mock import Mock, patch
+from chess_engine import game_state
+from enums import Player
+from Piece import Knight, Bishop
+import ai_engine
+
+
+class TestChess:
+    @pytest.fixture
+    def game(self):
+        return game_state()
+
+    @pytest.fixture
+    def board_empty(self):
+        game = game_state()
+        empty_row = [Player.EMPTY] * 8
+        game.board = [empty_row[:] for _ in range(8)]
+        return game
+
+    #Unit - test
+    def test1_get_valid_peaceful_moves(self, game):
+        knight = Knight('n', 3, 4, Player.PLAYER_1)
+        with patch.object(game, 'get_piece', return_value=Player.EMPTY):
+            knight_moves = knight.get_valid_peaceful_moves(game)
+            knight_moves_exccept = [(1, 3), (2, 2), (4, 2), (5, 3), (5, 5), (4, 6), (2, 6), (1, 5)]
+            assert set(knight_moves) == set(knight_moves_exccept)
+
+    def test2_knight_get_valid_peaceful_moves(self, game):
+        knight = Knight('n', 0, 0, Player.PLAYER_1)
+
+        def mock_get_piece(r, c):
+            if (r, c) != (0, 0) and (0 <= r < 8) and (0 <= c < 8):
+                return Player.EMPTY
+            else:
+                return None
+
+        game.get_piece = Mock(side_effect=mock_get_piece)
+        knight_moves = knight.get_valid_peaceful_moves(game)
+        knight_moves_exccept = [(1, 2), (2, 1)]
+        assert set(knight_moves) == set(knight_moves_exccept)
+
+    def test3_knight_get_valid_peaceful_moves(self, game):
+        knight = Knight('n', 7, 3, Player.PLAYER_1)
+
+        def mock_get_piece(r, c):

```

Comment:
seems as if you use this function several times in your code, as such it would make much more sense to make it a standalone method.